File: 2023/java/src/test/java/org/frieder/aoc/day17/TestDay17B.java
package org.frieder.aoc.day17;

import org.frieder.aoc.day17.b.Solution17B;
import org.junit.Test;

import java.io.IOException;

import static org.junit.Assert.assertEquals;

public class TestDay17B {

    @Test
    public void testExample1() throws IOException {
        assertEquals(94, Solution17B.getResult("../resources/17/test_a.txt"), 0);
    }

    @Test
    public void testExample2() throws IOException {
        assertEquals(71, Solution17B.getResult("../resources/17/test_b.txt"), 0);
    }


    @Test
    public void testResult() throws IOException {
        assertEquals(1411, Solution17B.getResult("../resources/17/input.txt"), 0);
    }
}

File: 2023/php/test/TestDay17B.php
<?php declare(strict_types=1);

$loader = require __DIR__ . '/../vendor/autoload.php';

use day17\b\Solution17B;
use PHPUnit\Framework\TestCase;
use utils\TestLogger;

final class TestDay17B extends TestCase
{

    public function testExample(): void
    {
        $this->assertSame(94, Solution17B::getResult("../resources/17/test_a.txt", TestLogger::createTestLogger()));
    }

    public function testExampleB(): void
    {
        $this->assertSame(71, Solution17B::getResult("../resources/17/test_b.txt", TestLogger::createTestLogger()));
    }

    public function testResult(): void
    {
        $this->assertSame(1411, Solution17B::getResult("../resources/17/input.txt", TestLogger::createTestLogger()));
    }

}